app/templates/book_details.html

```diff
@@ -19,15 +19,32 @@
                         </div>
                     </div>
                     <div class="col-8">
-                        <h3 class="card-title">{{ book.title }}</h3>
-                        <h6>{{ book.author }}</h6>
+                        <div class="row">
+                            <div class="col-10" id="booktitle">
+                                 <h3 class="card-title">{{ book.title }}</h3>
+                                                         <h6>{{ book.author }}</h6>
                         <p>{{book.rating}}
                             {% for star in rating_to_stars(book.rating) %}
                             <span>{{star}}</span>
                             {% endfor %}
                         </p>
+                            </div>
+                            <div class="col-2">
+                                    <div class="anim-icon anim-icon-md star">
+    <input type="checkbox" id="star1" />
+    <label for="star1"></label>
+  </div>
+</div>
+</div>
+
+
                         <p class="card-text">{{book.description|safe}}</p>
-                        <a href="{{book.amazon_url}}" target="_blank" class="btn btn-primary">Buy This Book</a>
+                        <a href="{{ book.amazon_url }}" class="btn btn-primary">Buy This Book</a>
```

Comment:
Can you check whether the `amazon_url` is set. if not, the button should be disabled and not clickable.